apps/media-viewer/src/Mediaviewer.vue

```diff
@@ -24,11 +24,143 @@
+      console.log('$_loader_loadFolder', contextRouteName, filePath)
+
+      // TODO: support public-files as well
+      if (this.$store.getters.activeFile.path === '' && contextRouteName === 'public-files') {
+        console.warn('TODO: handle public-files with passwords and everything, until then we redirect to the main public link page')
+        const path = this.$route.params.filePath.substring(1)
+        const token = path.substr(0, path.indexOf('/'))
+        this.$nextTick(() => {
+          console.log('wat')
+          this.$router.push({
+            name: 'publicLink',
+            params: {
+              token
+            }
+          })
+        })
+        throw new Error('public-files')
+      }
+
+      // load files
+      if (this.$store.getters.activeFile.path === '') {
+        const absolutePath = filePath.substring(1, filePath.lastIndexOf('/'))
+
+        return this.loadFolder({
+          client: this.$client,
+          absolutePath: absolutePath,
+          $gettext: this.$gettext,
+          routeName: contextRouteName,
+          loadSharesTree: !this.publicPage()
+        }).then(() => {
+          this.folderLoading = false
+        }).catch((error) => {
+          // FIXME: Loading of public link folders doesn't work at all and is disabled hence, so this code should be unreachable
+
+          // // password for public link shares is missing -> this is handled on the caller side
+          // if (this.$_loader_publicContext() && error.statusCode === 401) {
+          //   this.$router.push({
+          //     name: 'public-link',
+          //     params: {
+          //       token: this.$route.params.item
+          //     }
+          //   })
+          //   return
+          // }
+
+          this.showMessage({
+            title: this.$gettext('Loading folder failed…'),
+            desc: error.message,
+            status: 'danger'
+          })
+        })
+      }
+
+      // folder already loaded, nothing to do …
+      this.folderLoading = false
+    },
+
+    $_loader_getDavFilePath (filePath, query) {
+      let path = [
+        '..',
+        'dav',
+        'public-files',
+        filePath
+      ].join('/')
+
+      if (!this.$_loader_publicContext()) {
+        path = [
+          '..',
+          'dav',
+          'files',
+          this.$store.getters.user.id,
+          filePath
+        ].join('/')
+      }
+
+      return this.$client.files.getFileUrl(path) + '?' + queryString.stringify(query)
+    },
+
+    $_loader_headers () {
+      if (!this.$_loader_publicContext()) {
+        return null
+      }
+
+      const headers = new Headers()
+      headers.append('X-Requested-With', 'XMLHttpRequest')
+
+      const password = this.publicLinkPassword
+      if (password) {
+        headers.append('Authorization', 'Basic ' + Buffer.from('public:' + password).toString('base64'))
+      }
+      return headers
+    },
+
+    $_loader_navigateToContextRoute (contextRouteName, filePath) {
```

Comment:
For scrolling to the current file when going back to the files app you can do this:
`import { basename, dirname } from 'path'` and then
```
this.$router.push({
  name: contextRouteName,
  params: {
    item: dirname(filePath) || '/'
  },
  query: {
    scrollTo: basename(filePath)
  }
})
```
as the method body of `$_loader_navigateToContextRoute` (remove yours)

Reply:
Can you pls check if the scrolling works when the item is deep down in the list? There might be an issue with this related to https://github.com/owncloud/phoenix/issues/3265

Reply:
Yes I can. But fixing any of that is way out of scope for this PR. And I'd hope the API would stay stable no matter if theres virtual scrolling or pagination or whatever in place.

Reply:
Sure, just wanted to be safe and know how it behaves here. Absolutely no need for a fix here. And I agree that the API should stay stable. :+1:

Reply:
Alright, I tested it - I could not make it scroll anywhere :joy: 
What works tho: the file is selected, which is nice for starters.